clang/include/clang/Driver/Compilation.h

```diff
@@ -61,11 +61,11 @@ class Compilation {
       OrderedOffloadingToolchains;
 
   /// The original (untranslated) input argument list.
-  llvm::opt::InputArgList *Args;
+  std::unique_ptr<llvm::opt::InputArgList> Args;
 
   /// The driver translated arguments. Note that toolchains may perform their
   /// own argument translation.
-  llvm::opt::DerivedArgList *TranslatedArgs;
+  std::shared_ptr<llvm::opt::DerivedArgList> TranslatedArgs;
```

Comment:
This can be a unique_ptr as well; TranslatedArgs is fully owned by the Compilation class and doesn't share ownership with anyone else.